x/wasm/types/proposal.go

```diff
@@ -113,6 +113,16 @@ func (p StoreCodeProposal) ValidateBasic() error {
 			return sdkerrors.Wrap(err, "instantiate permission")
 		}
 	}
+
+	if len(p.Source) == 0 {
```

Comment:
As mentioned above, these fields should be optional on the server side to support a transition period for in flight proposals. 
But there is no problem to enforce checks on the CLI before submitting a proposal.
There can be stronger constraints on the expected data (when set).